backbone.js

```diff
@@ -14,24 +14,24 @@
 
   // Set up Backbone appropriately for the environment. Start with AMD.
   if (typeof define === 'function' && define.amd) {
-    define(['underscore', 'jquery', 'exports'], function(_, $, exports) {
+    define(['underscore', 'jquery', 'promise', 'exports'], function(_, $, Promise, exports) {
       // Export global even in AMD case in case this script is loaded with
       // others that may still expect a global Backbone.
-      root.Backbone = factory(root, exports, _, $);
+      root.Backbone = factory(root, exports, _, $, Promise);
     });
 
   // Next for Node.js or CommonJS. jQuery may not be needed as a module.
   } else if (typeof exports !== 'undefined') {
-    var _ = require('underscore'), $;
+    var _ = require('underscore'), $, Promise = require('promise');
     try { $ = require('jquery'); } catch(e) {}
-    factory(root, exports, _, $);
+    factory(root, exports, _, $, Promise);
 
   // Finally, as a browser global.
   } else {
-    root.Backbone = factory(root, {}, root._, (root.jQuery || root.Zepto || root.ender || root.$));
+    root.Backbone = factory(root, {}, root._, (root.jQuery || root.Zepto || root.ender || root.$), root.Promise);
   }
 
-}(function(root, Backbone, _, $) {
+}(function(root, Backbone, _, $, Promise) {
```

Comment:
I'm not a fan of requiring a particular Promise implementation. Many people will end up requiring both the library you have chosen here *and* whatever Promise library they actually use.

In my libs I prefer doing:

```js
if (typeof Promise !== 'undefined') {
  Backbone.Promise = Promise;
}
```

That way, if someone's using native Promises, and they wish to continue using native promises, they can. Or if they'd rather use RSVP, they can do that, too. And so on.

An extra install instruction – telling people to configure `Backbone.Promise` – is worth the added benefit of flexibility, I think.